src/AwsS3Adapter.php

```diff
@@ -438,7 +448,7 @@ public function listContents($dirname = '', $recursive = false)
      */
     protected function normalizeResponse(array $object, $path = null)
     {
-        $result = ['path' => $path ?: $this->removePathPrefix($object['Key'])];
+        $result = ['path' => $path ?: $this->removePathPrefix(isset($object['Key']) ? $object['Key'] : $object['Prefix'])];
```

Comment:
I'm not sure I follow... If the directory is stored as Prefix, then all the files would be named the same, which wouldn't be correct. Or is the file path returned under the Prefix key?